custom_components/bosch_alarm/lock.py

```diff
@@ -0,0 +1,76 @@
+"""Support for Bosch Alarm Panel points as binary sensors"""
+
+from __future__ import annotations
+
+import logging
+import re
+
+from homeassistant.components.lock import (
+    LockEntityFeature,
+    LockEntity,
+)
+
+from .const import DOMAIN
+
+_LOGGER = logging.getLogger(__name__)
+
+
+class PanelLockEntity(LockEntity):
+    def __init__(self, id, panel_conn, door):
+        self._observer = door.status_observer
+        self._panel = panel_conn.panel
+        self._attr_unique_id = f"{panel_conn.unique_id}_door_{id}"
+        self._attr_device_info = panel_conn.device_info()
+        self._attr_has_entity_name = True
+        self._door = door
+        self._door_id = id
+        self._attr_supported_features = LockEntityFeature.OPEN
+
+    @property
+    def name(self):
+        return self._door.name
+
+    @property
+    def is_open(self):
+        return self._door.is_open()
+
+    @property
+    def is_locked(self):
+        return self._door.is_locked()
+
+    @property
+    def available(self):
+        return self._door.is_open() or self._door.is_locked()
+
+    @property
+    def should_poll(self):
+        return False
+
+    async def async_lock(self):
+        await self._panel.door_relock(self._door_id)
+
+    async def async_unlock(self):
+        await self._panel.door_unlock(self._door_id)
+
+    async def async_open(self):
```

Comment:
I'm not sure I follow all the terminology here -- what's the difference between "open", "unclock", and "cycle"?
Is "cycle" an optional capability in Bosch locks? Maybe we should be checking if the lock supports it (command 0x0A?) and propagating this to HA with `LockEntityFeature`.

Reply:
Unlock unlocks the door and it stays unlocked
Lock locks an unlocked door
Cycle opens the door for a bit so that someone can walk through and then locks it again - essentially behaving like a keycard was just swiped
Every door would support cycling as thats the default behaviour when you swipe the keycard

Reply:
I was trying to understand by what the home assistant documentation means when they say that `open` operates the "latch", and I found [this](https://community.home-assistant.io/t/mqtt-lock-open/232823/10?u=mag1024) thread. Seems like the `open` is supposed to affect a different part of the lock, not just unlock/relock the main lock with a delay, so cycle may not be appropriate here.

We should probably not be implementing `async_open` or `is_open` here. A more appropriate way to expose the cycle functionality, if you think that's worthwhile, might be an integration service.

Reply:
Ah okay, that makes sense. We can just drop the cycle functionality for now and potentially add service commands later if someone requires that functionality.

Reply:
Have now removed the open feature.